src/source/source_cache.js

```diff
@@ -330,10 +332,10 @@ class SourceCache extends Evented {
      * @private
      */
     _retainLoadedChildren(
-        idealTiles: {[_: any]: OverscaledTileID},
+        idealTiles: {[number | string]: OverscaledTileID},
         zoom: number,
         maxCoveringZoom: number,
-        retain: {[_: any]: OverscaledTileID}
+        retain: {[number | string]: OverscaledTileID}
```

Comment:
Nit: let's stick to one type for a key if possible (`tile.key` is `number` so I think we can do away with `number`).

Reply:
Flow can't infer a number type as a key and there are many iterations over `this._tiles` https://flow.org/try/#0PTAEAEDMBsHsHcBQiDGsB2BnALqbBLaAU0wC5QBvAbXQFcBbAIyICcBdcgQ3QE8BfUAF5KfZJFgtQACjRZcufOjyESASkqJQW0LJzLiQ-SSrY2AbkSigA

```js
const tiles: {[number]: any} = {}

for (const t in tiles) {
    const tile = tiles[t];
}
```

```
const tile = tiles[t];
                   ^ string [1] is incompatible with number [2]. [incompatible-type]
References:
5: for (const t in tiles) {
              ^ [1]
3: const tiles: {[number]: any} = {}
                  ^ [2]
```